src/miner.cpp

```diff
@@ -125,8 +126,7 @@ CBlockTemplate* CreateNewBlock(const CChainParams& chainparams, const CScript& s
     pblocktemplate->vTxFees.push_back(-1); // updated at end
     pblocktemplate->vTxSigOps.push_back(-1); // updated at end
 
-    // Largest block you're willing to create:
-    unsigned int nBlockMaxSize = GetArg("-blockmaxsize", DEFAULT_BLOCK_MAX_SIZE);
+    // Largest block you're willing to create
     // Limit to betweeen 1K and MAX_BLOCK_SIZE-1K for sanity:
     nBlockMaxSize = std::max((unsigned int)1000, std::min((unsigned int)(MAX_BLOCK_SIZE-1000), nBlockMaxSize));
```

Comment:
It's a "global" variable that is used in exactly one function, which is the same function that you're looking at here (just down 100 lines or so), and part of the same critical section.

Anyway, if you don't like the way it looks, I can take a look at it again in a few days. Kinda busy with other stuff right now, though.